src/manifest.js

```diff
@@ -92,6 +93,21 @@ export const parseManifest = ({
     manifest.targetDuration = targetDuration;
   }
 
+  const parts = getLastParts(manifest);
+
+  if (parts.length && !manifest.partTargetDuration) {
+    let partTargetDuration = (manifest.targetDuration / parts.length);
+
+    if (manifest.segments && manifest.segments.length) {
+      partTargetDuration = parts.reduce((acc, p) => Math.max(acc, p.duration), 0);
+    }
```

Comment:
I think this logic was a little half baked here. Going to switch it so that `partTargetDuration` is just the largest part duration that we have at the time of parsing, as we will only add a partTargetDuration if we also have parts.